packages/curve-ui-kit/src/utils/array.ts

```diff
@@ -21,3 +21,36 @@ export function takeTopWithMin<T>(items: T[], getValue: (item: T) => number, thr
   const sorted = orderBy(items, getValue, 'desc')
   return takeWhile(sorted, (_item, index) => getValue(sorted[index]) >= threshold || index < minCount)
 }
+
+/**
+ * Computes a cutoff value that guarantees at least `minCount` items pass.
+ * - This function is the numeric counterpart of `takeTopWithMin`.
+ *
+ * Behavior:
+ * - If at least `minCount` items have value ≥ `threshold`, returns `threshold` unchanged.
+ * - Otherwise, returns the `minCount`-th largest value (the fallback cutoff so that ≥ `minCount` items pass).
+ *   If fewer than `minCount` items exist, returns the smallest value.
+ *
+ * This is useful to build a filter that prefers a fixed threshold, while
+ * still ensuring a minimum number of items are shown when the data is sparse.
+ *
+ * @template T - The type of items in the array.
+ * @param items - The array of items to inspect.
+ * @param getValue - Getter that maps an item to a numeric value.
+ * @param threshold - Preferred minimum value for passing items.
+ * @param minCount - Minimum number of items that must pass.
+ * @returns The cutoff value to use for filtering.
+ */
+export function minCutoffForTopK<T>(
+  items: T[],
+  getValue: (item: T) => number,
+  threshold: number,
+  minCount: number,
+): number {
+  const valuesDesc = orderBy(items.map(getValue), undefined, 'desc')
+  const firstBelowIdx = valuesDesc.findIndex((v) => v < threshold)
+  if (firstBelowIdx === -1 || firstBelowIdx >= minCount) return threshold
```

Comment:
`firstBelowIdx === -1` is unnecessary, unless you're expecting `minCount` to be -2 or less 😂